pybop/parameterisation.py

```diff
@@ -14,25 +13,48 @@ class Parameterisation:
     Parameterisation class for pybop.
     """
 
-    def __init__(self, model, observations, fit_parameters, x0=None, options=None):
-        self.model = model
+    def __init__(
+        self,
+        model,
+        observations,
+        fit_parameters,
+        geometry=None,
+        submesh_types=None,
+        var_pts=None,
+        spatial_methods=None,
+        solver=None,
+        x0=None,
+        check_model=True,
+        init_soc=None,
+    ):
+        self.model = model.pybamm_model
```

Comment:
seems confusing that you are storing the pybamm_model rather than the pybop one

Reply:
I would write `Parameterisation` assuming that you know nothing about pybamm. You have a pybop model which has a certain api, then just call functions on the pybop model (e.g. build_model()). The pybop model knows it stores a pybamm model, so any implementation details that depend on it being a pybamm model should go in the pybop model class